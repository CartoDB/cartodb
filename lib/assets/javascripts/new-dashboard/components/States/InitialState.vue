<template>
  <div class="initial-state">
    <div class="initial-state-icon">
      <slot name="icon"></slot>
    </div>
    <div class="initial-state-text">
      <h6 class="title is-medium is-txtGrey initial-state-title">{{title}}</h6>
      <slot name="description"></slot>
    </div>
    <div class="action-button">
      <slot name="actionButton"></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'InitialState',
  props: {
    title: String
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.initial-state {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
  padding: 36px;
  border: 2px dashed $grey;
  border-radius: 2px;
}

.initial-state-title {
  margin-bottom: 16px;
}

.initial-state-icon {
  margin-right: 36px;

  @media (max-width: $layout-tablet) {
    margin-bottom: 16px;

    img {
      width: 32px;
    }
  }
}

.initial-state-text {
  max-width: 480px;
}

.action-button {
  margin-left: auto;

  @media (max-width: $layout-tablet) {
    width: 100%;
    margin-top: 16px;
  }
}
</style>
