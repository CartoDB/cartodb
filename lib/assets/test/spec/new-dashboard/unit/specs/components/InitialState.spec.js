import { shallowMount } from '@vue/test-utils';
import InitialState from 'new-dashboard/components/States/InitialState';

describe('InitialState.vue', () => {
  it('should render correctly the initial state box', () => {
    const title = {
      title: 'My title'
    };
    const initialState = shallowMount(InitialState, {
      propsData: title,
      slots: {
        icon: '<img src="test.svg"/>',
        description: '<p>This is my description</p>',
        actionButton: '<button class="test">This is a button test</button>'
      }
    });

    expect(initialState).toMatchSnapshot();
  });
});
