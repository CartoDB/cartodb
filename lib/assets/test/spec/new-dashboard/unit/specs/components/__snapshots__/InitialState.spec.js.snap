// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialState.vue should render correctly the initial state box 1`] = `
<div class="initial-state">
  <div class="initial-state-icon"><img src="test.svg"></div>
  <div class="initial-state-text">
    <h6 class="title is-medium is-txtGrey initial-state-title">My title</h6>
    <p>This is my description</p>
  </div>
  <div class="action-button"><button class="test">This is a button test</button></div>
</div>
`;
